main.go

```diff
@@ -60,7 +60,7 @@ func main() {
 	// User handler
 	router.GET("user/:username", func(c *gin.Context) {
 
-		username := c.Param("username")
+		username := strings.ToLower(c.Param("username"))
```

Comment:
Ideally I would want to preserve the case of the username as sent by the user. The lowercase should only happen while searching for the cache file and while storing the cache file.